server/api.go

```diff
@@ -0,0 +1,407 @@
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"runtime/debug"
+
+	"github.com/gorilla/mux"
+	"github.com/mattermost/mattermost-server/v5/model"
+)
+
+// HTTPHandlerFuncWithUser is http.HandleFunc but userID is already exported
+type HTTPHandlerFuncWithUser func(w http.ResponseWriter, r *http.Request, userID string)
+
+// ResponseType indicates type of response returned by api
+type ResponseType string
+
+const (
+	// ResponseTypeJSON indicates that response type is json
+	ResponseTypeJSON ResponseType = "JSON_RESPONSE"
+	// ResponseTypePlain indicates that response type is text plain
+	ResponseTypePlain ResponseType = "TEXT_RESPONSE"
+	// ResponseTypeDialog indicates that response type is a dialog response
+	ResponseTypeDialog ResponseType = "DIALOG"
+)
+
+type APIErrorResponse struct {
+	ID         string `json:"id"`
+	Message    string `json:"message"`
+	StatusCode int    `json:"status_code"`
+}
+
+func (p *Plugin) initializeAPI() {
+	p.router = mux.NewRouter()
+	p.router.Use(p.withRecovery)
+
+	apiRouter := p.router.PathPrefix("/api/v1").Subrouter()
+
+	apiRouter.HandleFunc("/game/{gameID}/flip", p.extractUserMiddleWare(p.handleFlipCard, ResponseTypeJSON)).Methods(http.MethodPost)
```

Comment:
Same comment as before in the other plugin, I think we can use the `extractUserMiddleware` as a real middleware.